pkg/util/secret.go

```diff
@@ -86,6 +86,9 @@ func PatchSecret(client kubeclient.Interface, namespace, name string, pt types.P
 
 // CreateOrUpdateSecret creates a Secret if the target resource doesn't exist. If the resource exists already, this function will update the resource instead.
 func CreateOrUpdateSecret(client kubeclient.Interface, secret *corev1.Secret) error {
+	// Fix that resourceVersion should not be set on objects to be created
+	secret.ResourceVersion = ""
+
```

Comment:
The first issue I found is the `CreateOrUpdateSecret` probably does not work as expected, I mean we expect it can fallback to `Update` once encountering the `AlreadyExist` error during creating, but since it doesn't take the latest resource version, so the `Update` would fail again.

For example, the [CreateOrUpdate](https://github.com/kubernetes-sigs/controller-runtime/blob/4c9c9564e4652bbdec14a602d6196d8622500b51/pkg/controller/controllerutil/controllerutil.go#L195) implemented by controller-runtime, it always [deep-copy](https://github.com/kubernetes-sigs/controller-runtime/blob/4c9c9564e4652bbdec14a602d6196d8622500b51/pkg/controller/controllerutil/controllerutil.go#L210) the existing object(take latest resource version) and then make changes(`mutate`) before try updation.

Reply:
> but since it doesn't take the latest resource version, so the Update would fail again.

It seems not to happen.

Reply:
> but since it doesn't take the latest resource version, so the `Update` would fail again

Yeah, `Update` will fail because of old resourceVersion. Caller shall refetch the object and retry.

Reply:
> It seems not to happen.

After client fetch the object, other client update it. Then update will fail.

Reply:
That's right. I created #2687 for tracking this.